lib/logstash/outputs/elasticsearch/common_configs.rb

```diff
@@ -75,13 +75,8 @@ def self.included(mod)
       # to prevent LS from sending bulk requests to the master nodes.  So this parameter should only reference either data or client nodes in Elasticsearch.
       mod.config :hosts, :validate => :array, :default => ["127.0.0.1"]
 
-      # This plugin uses the bulk index API for improved indexing performance.
-      # This setting defines the maximum sized bulk request Logstash will make
-      # You you may want to increase this to be in line with your pipeline's batch size.
-      # If you specify a number larger than the batch size of your pipeline it will have no effect,
-      # save for the case where a filter increases the size of an inflight batch by outputting
-      # events.
-      mod.config :flush_size, :validate => :number, :default => 500
+      # DEPRECATED. We now restrict bulk sizes to a size that balances heap size with reducing round-trips. Very large events will be sent in a single request by themselves.
+      mod.config :flush_size, :validate => :number, :default => 500, :deprecated => "No Longer Needed"
```

Comment:
@jsvd my original intent with `deprecate` was, at the configuration interface, that a configuration would still be loaded successfully but we would warn a user about the deprecation. With obsolete, my intent was to fail to load the config but tell the user why.

From this, I think making it :deprecated as a no-op is fine, since the config still runs.

You have a good point about the folks setting flush_size previously, though, but to be honest, all the uses I can think of are basically workarounds for perceived for actual problems with how much we send to Elasticsearch in a single request. This patch hopefully will remove those problems (perceived or actual) with Elasticsearch bulk sizes, and in doing so, will remove the need for that band-aid -- therefore, I think it's OK to immediately turn :flush_size into a no-op.

Thoughts? Am I missing a key reason that isn't a work-around for why someone would set flush_size?